include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_radix_sort.h

```diff
@@ -605,7 +605,7 @@ struct __parallel_radix_sort_iteration
         // correct __block_size according to local memory limit in count phase
         const auto __max_count_wg_size = oneapi::dpl::__internal::__slm_adjusted_work_group_size(
             __exec, sizeof(typename __decay_t<_TmpBuf>::value_type) * __radix_states, __block_size);
-        __block_size = oneapi::dpl::__internal::__dpl_ceiling_div(__max_count_wg_size, __radix_states);
+        __block_size = oneapi::dpl::__internal::__dpl_ceiling_div(__max_count_wg_size, __radix_states) * __radix_states;
```

Comment:
Let's do "floor" division instead of "ceil" division.

```suggestion
        __block_size = static_cast<::std::size_t>((__max_count_wg_size / __radix_states)) * __radix_states;
```

 `__slm_adjusted_work_group_size` returns the maximum number of work-items in a work-group according to SLM usage per each work-item. It is `__max_count_wg_size`. With "ceil" division, we may end up with `__block_size` bigger than `__max_count_wg_size` and more SLM allocated than the device allows.

Reply:
Yeah, good point. I guess in that case we might also want to assert that we don't end up with a `__block_size` of zero?

Reply:
Thanks for noticing it. I would suggest leaving it as is right now. Unfortunately, I do not see an easy way to handle that case properly. I've created #977 with a suggestion to implement a fallback implementation of count kernel which does not use SLM.

Line 612 will correct `0` to `__radix_bits`. However, If we get `0`, it will indicate that we do not have enough SLM. I expect that SYCL backend runtime will throw an exception about not having enough resources.